enhancements/security/api-tls-curves-config.md

```diff
@@ -0,0 +1,164 @@
+
+The ability to set curves explicitely will also make it possible to align our 
+OpenShift TLS profiles to match the curves present in the [Mozilla TLS Profiles](https://wiki.mozilla.org/Security/Server_Side_TLS). 
+
+This change will require working with OpenShift component owners to use this new field, however this is outside the scope of this proposal.
+
+### Workflow Description
+
+Administrators will use the [existing custom TLS security profile flow](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/security_and_compliance/tls-security-profiles#tls-profiles-ingress-configuring_tls-security-profiles) for setting the supported curves. 
+
+Specifically administrators will use 
+
+`oc edit IngressController default -n openshift-ingress-operator`
+
+and edit the spec.tlsSecurityProfile field:
+
+```
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+ ...
+spec:
+  tlsSecurityProfile:
+    type: Custom 
+    custom: 
+      ciphers: 
+      - ECDHE-RSA-CHACHA20-POLY1305
+      minTLSVersion: VersionTLS13
+      curves:
+      - X25519MLKEM512
+ ...
+```
+
+### API Extensions
+
+- Adds a `curves` field to the `spec.tlsSecurityProfile` 
+- The addition of this field should not affect existing API behaviour
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+
+#### Standalone Clusters
+
+N/A
+
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+
+### Implementation Details/Notes/Constraints
+
+#### Mismatching curves and ciphersuites
+There is a case where the administrator could incorrectly specificy a set of ciphersuites
+that do not work with each other. For example using an RSA ciphersuite with a ECDHE curve (such as TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 and P-256). The default behavior OpenSSL as well as go's crypto/tls (both used extensively in OpenShift) is to fail at **TLS handshake time**. . The TLS server instance will start normally, but when TLS clients attempt to handshake with the TLS server, the handshake will fail with a `handshake failure`
+
+I propose that this is the *desired* behavior of OpenShift. Administrators that utilize the custom TLS profile are foregoing the guaranteed correctly configured TLS profiles (such as Modern, Intermediate, etc.) and the system should comply accordingly. The landscape of TLS is constantly evolving and relying on the TLS implementation (Openssl, golang crypto/tls, etc) itself provides users more flexibility. Adding a validation system on top as part of the OpenShift would be cumbersome and restricts admins to use what we deem compatible rather than what the underylying TLS implementation is capable of.
```

Comment:
>Adding a validation system on top as part of the OpenShift would be cumbersome and restricts admins to use what we deem compatible rather than what the underylying TLS implementation is capable of

We are generally expected to be the SMEs here. If there are combinations that we know ahead of time will not work, why should we not prevent users from doing that earlier in the loop?

Where possible we should provide appropriate guardrails to help prevent users from breaking their own clusters.